### DISCUSSION TITLE
Squash Merge - no commit message draft?

### DISCUSSION THREAD

**Discussion by Jabbado** [gitextensions/gitextensions#10033] [Q&A]
When doing a squash merge (check "Squash commits" option) the commit dialog when actually committing the changes is empty. This is unlike the git console which pops up the text editor when doing the actual commit so you can edit the message(s) before doing the actual commit.

Is there a way to do this without just copy and pasting the messages myself?

**Comment by vbjay** [comment]
You don't merge in commit dialog. In merge dialog in advanced section there
is an edit commit message option.

**Reply by Jabbado** [reply]
Ok but how can this be auto-filled with a draft containing all the commit messages to be squashed?

**Comment by RussKie** [comment]
Some screenshots would be useful in understanding what and where are doing
this.

**Reply by Jabbado** [reply]
![Squash-Merge](https://user-images.githubusercontent.com/39350939/173163609-3907ceb0-56ce-423e-a9ed-44bfe0b49e15.png)

The highlighted option above tells it to do a squash merge doesn't it?

In the console we can do like:
```
git checkout main
git merge --squash feature
git commit
```
This will bring up the text editor auto-filled with the messages from all the commits about to be squashed, so they can be edited. There's no need to copy and paste each message manually - just edit and save to our liking. 

Can GitExtensions do the same?

**Reply by mstv** [reply]
`git merge --squash ...` stores descriptions of the squashed commits in `.git/SQUASH_MSG`.
The GE Commit dialog is not aware of this file as it is of `COMMITMESSAGE` and `MERGE_MSG`.

I see two options:
1. GE could run `git commit` after _successful_ `git merge --squash ...`. Git opens the editor with a pre-filled commit message.
2. The Commit dialog could read `.git/SQUASH_MSG`.

For the time being, you could create a toolbar button for `git commit` by means of a user script:

![image](https://user-images.githubusercontent.com/36601201/173176233-6b20fb3f-1b6b-4ca1-8cd7-bdbe33650ad6.png)

**Reply by Jabbado** [reply]
Ok that's great! Thanks. Is there any docs for the scripting system? I was wondering if its possible to chain commands so a script would first run
`git merge --squash`
on the selected branch then run 
`git commit` 
straight after.

**Reply by mstv** [reply]
https://git-extensions-documentation.readthedocs.io/en/master/settings.html#git-extensions-scripts

```
Command: cmd
Arguments: /c git merge --squash {sBranch} && git commit
```

**Reply by Jabbado** [reply]
Thank you but will that command run the merge on the currently selected branch?

**Reply by mstv** [reply]
![image](https://user-images.githubusercontent.com/36601201/173177887-af723112-2503-4d3b-8ceb-3f7b975e0656.png)

**Reply by Jabbado** [reply]
Thank you for the help, @mstv. This works a treat. [Your reply](https://github.com/gitextensions/gitextensions/discussions/10033#discussioncomment-2927443) should be the answer but I can't mark it as such. Maybe a mod can do it?

**Comment by vbjay** [comment]
https://askubuntu.com/questions/334994/which-one-is-better-using-or-to-execute-multiple-commands-in-one-line.
Covers
bash  command chaining. Note you would need to alter from git to running
bash and calling git if multiple commands. It is doable.
